addons/sms/wizard/sms_composer.py

```diff
@@ -237,12 +237,16 @@ def _action_send_sms_mass(self, records=None):
         records = records if records is not None else self._get_records()
 
         sms_record_values = self._prepare_mass_sms_values(records)
-        sms_all = self._prepare_mass_sms(records, sms_record_values)
 
-        if sms_all and self.mass_keep_log and records and issubclass(type(records), self.pool['mail.thread']):
+        if sms_record_values and self.mass_keep_log and records and issubclass(type(records), self.pool['mail.thread']):
             log_values = self._prepare_mass_log_values(records, sms_record_values)
-            records._message_log_batch(**log_values)
+            mail_messages = records._message_log_batch(**log_values)
+            for mail_message in mail_messages:
+                sms_record_vals = sms_record_values.get(mail_message.res_id)
+                if sms_record_vals:
```

Comment:
is it possible that this condition is False ?

Reply:
Yes I come to the same conclusion so I have updated the code.
Note that everything could be simplified in one line but I have kept the 2 operations: _prepare_mass_log_values and _message_log_batch in separate lines to make things clearer (using 2 variables only used once).